src/main/java/com/denizenscript/depenizen/bukkit/commands/noteblockapi/NBSCommand.java

```diff
@@ -62,72 +64,55 @@ public NBSCommand() {
-            }
-            else if (!scriptEntry.hasObject("file")
-                    && arg.matchesPrefix("file")) {
-                scriptEntry.addObject("file", arg.asElement());
-            }
-            else if (!scriptEntry.hasObject("action")
-                    && arg.matchesEnum(Action.class)) {
-                scriptEntry.addObject("action", arg.asElement());
-            }
-            else {
-                arg.reportUnhandled();
-            }
-        }
-        if (!scriptEntry.hasObject("action")) {
-            throw new InvalidArgumentsException("Action not specified! (play/stop)");
-        }
-        else if (!scriptEntry.hasObject("targets")) {
+    public static void autoExecute(ScriptEntry scriptEntry,
+                                   @ArgName("action") Action action,
+                                   @ArgName("file") @ArgPrefixed @ArgDefaultNull String file,
+                                   @ArgName("targets") @ArgPrefixed @ArgDefaultNull @ArgSubType(PlayerTag.class) List<PlayerTag> targets,
+                                   @ArgName("at") @ArgPrefixed @ArgDefaultNull LocationTag location,
+                                   @ArgName("distance") @ArgPrefixed @ArgDefaultText("16") int distance) {
+        if (targets == null) {
             if (Utilities.entryHasPlayer(scriptEntry)) {
-                scriptEntry.addObject("targets", Collections.singletonList(Utilities.getEntryPlayer(scriptEntry)));
+                targets = List.of(Utilities.getEntryPlayer(scriptEntry));
             }
             else {
-                throw new InvalidArgumentsException("Must specify players to add, remove or spectate!");
+                throw new InvalidArgumentsRuntimeException("Must specify players to add or remove!");
             }
         }
-    }
-
-    @Override
-    public void execute(ScriptEntry scriptEntry) {
-        ElementTag file = scriptEntry.getObjectTag("file");
-        ElementTag action = scriptEntry.getObjectTag("action");
-        List<PlayerTag> targets = (List<PlayerTag>) scriptEntry.getObject("targets");
-        if (scriptEntry.dbCallShouldDebug()) {
-            Debug.report(scriptEntry, getName(), action, db("targets", targets), file);
-        }
-        if (targets == null || targets.isEmpty()) {
-            Debug.echoError(scriptEntry, "Targets not found!");
-            return;
-        }
-        if (action.asString().equalsIgnoreCase("play")) {
-            if (file == null) {
-                Debug.echoError(scriptEntry, "File not specified!");
-                return;
+        switch (action) {
+            case PLAY -> {
+                if (file == null) {
+                    Debug.echoError(scriptEntry, "File not specified!");
+                    return;
+                }
+                String directory = URLDecoder.decode(System.getProperty("user.dir"));
+                Song s = NBSDecoder.parse(new File(directory + "/plugins/Denizen/" + file + ".nbs"));
```

Comment:
This is an issue with the original code not this PR, but this file handling is wrong - it should look more like
```java
File file = new File(Denizen.getInstance().getDataFolder(), fName + ".nbs");
if (!Utilities.canReadFile(file)) {
    Debug.echoError("Cannot read from that file path due to security settings in Denizen/config.yml.");
    return;
}
```